library/src/test/java/com/mapbox/android/gestures/MultiFingerTapGestureDetectorTest.java

```diff
@@ -0,0 +1,184 @@
+package com.mapbox.android.gestures;
+
+import android.view.MotionEvent;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import java.util.HashMap;
+
+import static junit.framework.Assert.assertFalse;
+import static junit.framework.Assert.assertTrue;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class MultiFingerTapGestureDetectorTest extends AbstractGestureDetectorTest {
+
+  private MultiFingerTapGestureDetector multiFingerTapGestureDetector;
+  @Mock
+  private MultiFingerTapGestureDetector.OnMultiFingerTapGestureListener listener;
+
+  private MotionEvent downMotionEvent;
+  private MotionEvent pointerDownMotionEvent;
+  private MotionEvent pointerUpMotionEvent;
+  private MotionEvent upMotionEvent;
+  private MotionEvent moveMotionEvent;
+
+  @Before
+  public void setUp() throws Exception {
+    super.setUp();
+    MockitoAnnotations.initMocks(this);
+    multiFingerTapGestureDetector = Mockito.spy(new MultiFingerTapGestureDetector(context, androidGesturesManager));
+    multiFingerTapGestureDetector.setListener(listener);
+    doReturn(true).when(multiFingerTapGestureDetector).checkPressure();
+    doReturn(false).when(multiFingerTapGestureDetector).isSloppyGesture();
+  }
+
+  @Test
+  public void twoFingerTapTest() {
+    dispatchMultiFingerTap(false);
```

Comment:
this kind of test feels weird. a test mostly reads like:

```
@Test
public void testSomething(){
  Object expected = {something};
  Object actual = {something};
   assertEquals(expected, actual);
}
```

a test should assert if something is equal, verify if a mocked object is called and that should be obvious from the test code itself. Don't worry about repeating code, that is allowed with tests.